### PR TITLE
feat(CI) add CodeQL Analysis

### PR DESCRIPTION
**Pull request by FrancescoBorzi**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
```
Comment (Kitzunu) [inline .github/workflows/codeql-analysis.yml:50]: Since it fails on autobuild

Comment (Kitzunu) [reply .github/workflows/codeql-analysis.yml:50]: https://github.com/azerothcore/azerothcore-wotlk/pull/3982/checks?check_run_id=1543143174

Comment (FrancescoBorzi) [reply .github/workflows/codeql-analysis.yml:50]: yes

**Comment by locus313** [comment]
This looks to be working now. results are getting posted. You can see the ones posted from this pr here:
https://github.com/azerothcore/azerothcore-wotlk/security/code-scanning?query=tool%3ACodeQL+ref%3Arefs%2Fpull%2F3982%2Fmerge

**Comment by Kitzunu** [comment]
Cool!

**Comment by Kitzunu** [comment]
@locus313 @FrancescoBorzi TBM?

**Comment by locus313** [comment]
converted to use the composite action. Guess the open question is do we want to leave the schedule in place or only run on push and pull request?

**Comment by FrancescoBorzi** [comment]
> converted to use the composite action. Guess the open question is do we want to leave the schedule in place or only run on push and pull request?

at the moment I don't really have an opinion myself on that (honestly @Kitzunu mostly worked to finish this PR), we can try push/pull and see how it goes

**Comment by FrancescoBorzi** [comment]
sorry of course I meant @Kitzunu and @locus313

**Comment by locus313** [comment]
gotcha. I just commented out the schedule part for now.

**Review by IntelligentQuantum — APPROVED** [review]

**Pull request closed without merging**